planner/core/exhaust_physical_plans.go

```diff
@@ -556,6 +556,8 @@ func (p *LogicalJoin) buildIndexJoinInner2TableScan(
 		}
 		pkMatched = true
 		keyOff2IdxOff[i] = 0
+		// There is only one EqualCondition for IndexJoin to construct Range when it is a primary key. For issue #14822.
```

Comment:
See the comment I left before. The range could be decided by more than one EqualCondition, only if these conditions contain inner primary key. For example: `select * from t1, t2 where t1.a = t2.pk and t1.b = t2.pk`, in this query, both `t1.a` and `t1.b` shoule be in `outerJoinKeys`.

Reply:
Sorry for the misread. I have change it now.